src/shims/posix/fs.rs

```diff
@@ -49,6 +50,29 @@ impl FileDescriptor for FileHandle {
         assert!(communicate_allowed, "isolation should have prevented even opening a file");
         Ok(self.file.seek(offset))
     }
+
+    fn close<'tcx>(&mut self, communicate_allowed: bool) -> InterpResult<'tcx, io::Result<i32>> {
+        assert!(communicate_allowed, "isolation should have prevented even opening a file");
+        // We sync the file if it was opened in a mode different than read-only.
+        if self.writable {
+            // `File::sync_all` does the checks that are done when closing a file. We do this to
+            // to handle possible errors correctly.
+            let result = self.file.sync_all().map(|_| 0i32);
+            // Now we actually close the file.
+            drop(self);
```

Comment:
Here, `self` has type `&mut FileDescriptor`. Dropping that reference does nothing.

I think `close` should take `self` by-value; it should consume, not borrow.